src/tnlp.hxx

```diff
@@ -369,18 +367,14 @@ namespace roboptim
 
       Eigen::Map<const typename function_t::argument_t> x_ (x, n);
 
-      typedef typename solver_t::problem_t::constraints_t::const_iterator
+      typedef typename std::vector<functionPtr_t>::const_iterator
 	citer_t;
 
       typename function_t::size_type idx = 0;
-      for (citer_t it = solver_.problem ().constraints ().begin ();
-	   it != solver_.problem ().constraints ().end (); ++it)
+      for (citer_t it = constraintFunctions_.begin ();
+	   it != constraintFunctions_.end (); ++it)
 	{
-	  shared_ptr<typename solver_t::commonConstraintFunction_t> g;
-	  if (it->which () == LINEAR)
-	    g = get<shared_ptr<linearFunction_t> > (*it);
-	  else
-	    g = get<shared_ptr<nonLinearFunction_t> > (*it);
+	  const function_t* g = &*(*it);
```

Comment:
Nope. Everywhere definitions like this remain some cleaning can still be done. I will do it